repository_pool.go

```diff
@@ -144,17 +148,23 @@ func (r *sivaRepository) Path() string {
 	return r.path
 }
 
+func (r *sivaRepository) Cache() cache.Object {
+	return r.cache
+}
+
 // RepositoryPool holds a pool git repository paths and
 // functionality to open and iterate them.
 type RepositoryPool struct {
 	repositories map[string]repository
 	idOrder      []string
+	cache        cache.Object
 }
 
-// NewRepositoryPool initializes a new RepositoryPool
-func NewRepositoryPool() *RepositoryPool {
+// NewRepositoryPool initializes a new RepositoryPool with LRU cache.
+func NewRepositoryPool(maxCacheSize cache.FileSize) *RepositoryPool {
 	return &RepositoryPool{
 		repositories: make(map[string]repository),
+		cache:        cache.NewObjectLRU(maxCacheSize),
```

Comment:
If I understand correctly we are using the same cache for all repositories. While this may be OK now we may have problems when partitions are used:

* I'm not really sure of the cache implementation thread safety
* Recent objects from one repo may be evicted by others and make it run slower
* Concurrent use of the same cache may be slower (locking?)

I would have one cache per repo, a pool of caches that can be evicted or one cache per partition. I think we can have one per repo for now and get back to it when partitions are in use.

Reply:
the problem of having one cache per repo is that we will not able to know the amount of memory gitbase will use. Maybe we can use as a default value 96 MiB * number of repositories. Or apart from that, maybe we can implement an LRU with two key layers, to evict keys from a specific repository. WDYT?

Reply:
So maybe we can keep one cache in a pool but add mutex and prefix keys by repo id?